app/controllers/user_profiles_controller.rb

```diff
@@ -0,0 +1,56 @@
+class UserProfilesController < ApplicationController
+  before_action :set_user, only: [:show, :edit, :update, :destroy]
+  
+  def index
+    @user_profile = UserProfile.all.order(id: :asc)
+  end
+
+  def show
+    @user_profile = UserProfile.find(params[:id])
+    @avatarsrc = "avatar/#{@user_profile.avatar_file_name}"
+  end
+
+  def edit
+    @user_profile = UserProfile.find(params[:id])
+    @avatarsrc= "avatar/#{@user_profile.avatar_file_name}"
+  end
+
+  def new
+    @user_profile = UserProfile.new
+    @avatarsrc = "avatar/#{@user_profile.avatar_file_name}"
+  end
+
+  def destroy
+    redirect_to user_profiles_path  if @user_profile.destroy 
+  end
+
+  def create
+    @user_profile = UserProfile.new(profile_params)
+    if @user_profile.save 
+      #render :text => "<script>alert('Profile Updated successfully')</script>"
+      redirect_to user_profile_path(@user_profile)
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @user_profile.update(profile_params)  
+      #render :text => "<script>alert('Profile Updated successfully')</script>"
```

Comment:
I prefer update_attributes method here. 
http://railsless.blogspot.in/2012/04/what-is-different-in-updateall-update.html